.github/actions/build-obs/action.yaml

```diff
@@ -0,0 +1,109 @@
+name: 'Setup and build obs-studio'
+description: 'Builds obs-studio for specified architecture and build config.'
+inputs:
+  target:
+    description: 'Build target for dependencies'
+    required: true
+  config:
+    description: 'Build configuration'
+    required: false
+    default: 'RelWithDebInfo'
+  codesign:
+    description: 'Enable codesigning (macOS only)'
+    required: false
+    default: 'false'
+  codesignIdent:
+    description: 'Developer ID for application codesigning (macOS only)'
+    required: false
+    default: '-'
+  codesignTeam:
+    description: 'Team ID for application codesigning (macOS only)'
+    required: false
+    default: ''
+  workingDirectory:
+    description: 'Working directory for packaging'
+    required: false
+    default: ${{ github.workspace }}
+runs:
+  using: composite
+  steps:
+    - name: Run macOS Build
+      if: runner.os == 'macOS'
+      shell: zsh --no-rcs --errexit --pipefail {0}
+      working-directory: ${{ inputs.workingDirectory }}
+      env:
+        CODESIGN_IDENT: ${{ inputs.codesignIdent }}
+        CODESIGN_TEAM: ${{ inputs.codesignTeam }}
+      run: |
+        : Run macOS Build
+
+        local -a build_args=(
+          --config ${{ inputs.config }}
+          --target macos-${{ inputs.target }}
+        )
+        if (( ${+RUNNER_DEBUG} )) build_args+=(--debug)
+
+        if [[ '${{ inputs.codesign }}' == true ]] build_args+=(--codesign)
+
+        git fetch origin --no-tags --no-recurse-submodules -q
+        .github/scripts/build-macos ${build_args}
+
+    - name: Install Dependencies 🛍️
+      if: runner.os == 'Linux'
+      shell: bash
+      run: |
+        : Install Dependencies 🛍️
+        echo ::group::Install Dependencies
+        eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
+        brew install --quiet zsh
```

Comment:
Maybe its better to install `zsh` via `apt`
```
apt-get install -y zsh > /dev/null
```

Reply:
This is on purpose - we manage pinned `clang-format` versions via Brew already and `apt` packages are usually severely out of date (especially on Ubuntu 22.04, which is still the most recent Linux runner version available on GitHub Actions).

On local machines, you can use any package manager of your choice to install some dependencies, but on CI it's easier for us to use Linuxbrew (which is also distribution-independent).

Reply:
Sounds reasonable, but the Linux build CI currently has an error:
`Error: zsh: command not found`
I think it is because it was not linked to the right place.

Reply:
Yeah the branch is in a non-working state - I’m currently testing on my own fork because of all the different event types and moving parts. There are far more issues in its current state many of which are fixed already.

Reply:
```
sudo update-alternatives --install /bin/zsh zsh /home/linuxbrew/.linuxbrew/bin/zsh 100
sudo update-alternatives --set zsh /home/linuxbrew/.linuxbrew/bin/zsh
```

Reply:
That's not necessary - you just have to add the paths to the `GITHUB_PATH` so that it's persisted for the current runner (because every step in a workflow has its own state and `export`s have no effect).